var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -53,12 +53,12 @@ class FoamExtend(Package):
     and owner of the OPENFOAM trademark.
     """
 
-    homepage = "http://www.extend-project.de/"
+    homepage = "https://sourceforge.net/projects/foam-extend/"
 
+    version("5.0", git="http://git.code.sf.net/p/foam-extend/foam-extend-5.0.git")
     version("4.1", git="http://git.code.sf.net/p/foam-extend/foam-extend-4.1.git")
     version("4.0", git="http://git.code.sf.net/p/foam-extend/foam-extend-4.0.git")
     version("3.2", git="http://git.code.sf.net/p/foam-extend/foam-extend-3.2.git")
-    version("3.1", git="http://git.code.sf.net/p/foam-extend/foam-extend-3.1.git")
```

Comment:
We generally recommend deprecating versions before removing them.  Also, why remove 3.1 and not 3.0?  (Could use a comment to explain.)

Deprecation of old versions is discussed in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions.

Reply:
The first reason is that you cannot download the file from the address of 3.1. It is unclear whether the file was originally non-existent or has been deleted, but currently, there is no repository at the address of 3.1. 

To provide further clarification, there are two sourceforge projects related to *foam-extend*.
1. https://sourceforge.net/projects/openfoam-extend/
2. https://sourceforge.net/projects/foam-extend/

The first project includes versions 3.0 and 3.1, while the second project encompasses versions 3.0, 3.2, 4.0, 4.1, and 5.0.
I consider the first project to be deprecated, which is why I based my evaluation on the second project.

If I refer to the guide, in reality, even versions 3.0 and 3.2 are quite outdated and not commonly used. It appears that they would require patches to work with the current compiler versions, so considering them as deprecated seems accurate to me. What are your thoughts on this?

Reply:
You can certainly mark them as deprecated then remove them after the next release.